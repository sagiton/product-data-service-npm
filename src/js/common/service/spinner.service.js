(function (angular) {
    angular
        .module('pds.common.service')
        .service('SpinnerService', SpinnerService);

    SpinnerService.$inject = ['$http'];

    function SpinnerService($http) {
        var self = this;

        self.isLoading = function () {
            return $http.pendingRequests.length > 0;
        }

    }

})(angular);
