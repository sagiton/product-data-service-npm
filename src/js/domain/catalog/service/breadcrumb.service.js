(function(angular) {
    angular
        .module("pds.catalog.service")
        .service("breadcrumbService", BreadcrumbService);

    BreadcrumbService.$inject = ['CatalogService', '_', '$q'];

    function BreadcrumbService(CatalogService, _, $q) {
        var templatePromise

        return {
            getData: getData,
            build: build
        };

        function getData(categoryId) {
            return templatePromise = templatePromise || CatalogService.getTemplate(categoryId, 'BREADCRUMBS')
        }

        function build(categoryId) {
            return getData(categoryId)
                .then(decorateWithUrls)
                .then(mapData);
        }

        function decorateWithUrls(response) {
            var tree = response.nodes
            var promises = _.map(tree, function (node) {
                return CatalogService
                    .resolveUriFromHierarchy(node.id)
                    .then(function (url) {
                        node.url = url;
                        return node
                    });
            })
            return $q.all(promises)
        }

        function mapData(tree) {
            return _.map(tree, function (node) {
                return _.pick(node, 'id', 'name', 'url', 'type')
            })
        }
    }

})(angular);
