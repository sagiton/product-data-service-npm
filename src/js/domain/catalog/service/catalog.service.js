(function (angular) {
    angular
        .module("pds.catalog.service")
        .service("CatalogService", CatalogService);

    CatalogService.$inject = ['$window', 'Catalog', 'MenuService', 'catalogUrlSchema', 'catalogSearchListener', '_', '$q', 'locale', 'metaTag'];

    function CatalogService($window, Catalog, menuService, catalogUrlSchema, catalogSearchListener, _, $q, locale, metaTag) {
        var catalogTemplate;

        catalogSearchListener
            .listen()
            .then(function (params) {
                return resolveUriFromHierarchy(params.target.resourceId, null, params.consumer);
            })
            .then(function (uri) {
                $window.location.href = uri;
            });

        return {
            getNewProducts: getNewProducts,
            getCatalogTemplate: getCatalogTemplate,
            getTemplate: getTemplate,
            redirectTo: redirectTo,
            travelUpNavigationHierarchy: travelUpNavigationHierarchy,
            getIdFromLocation: getIdFromLocation,
            resolveUriFromHierarchy: resolveUriFromHierarchy,
        };

        function getNewProducts() {
            var catalog = new Catalog({
                template: {name: 'NEW_PRODUCTS'},
                model: {
                    locale: locale.toString(),
                    channel: metaTag.getOcsChannel()
                }
            });
            return catalog.$template();
        }

        function getCatalogTemplate(catalogId) {
            catalogTemplate = catalogTemplate || getTemplate(catalogId)
            return catalogTemplate
        }

        function getTemplate(catalogId, type) {
            return menuService.findInNavigation(catalogId)
                .then(function (catalog) {
                    catalog.template = {
                        name: type || catalog.type
                    }
                    catalog.model = {
                        locale: locale.toString(),
                        channel: metaTag.getOcsChannel(),
                        catalogRequest: {
                            id: catalogId,
                            channel: metaTag.getOcsChannel(),
                            type: catalog.type
                        }
                    }
                    return catalog.$template();
                });
        }

        function redirectTo(id) {
            return resolveUri(id)
                .then(function (uri) {
                    $window.location.href = uri;
                });
        }

        function travelUpNavigationHierarchy(categoryId, locale) {
            return menuService
                .findInNavigation(categoryId, locale)
                .then(function (catalog) {
                    var catalogs = [];
                    while (catalog != null) {
                        catalogs.push(catalog);
                        catalog = menuService.findParentInNavigation(catalog.id, locale);
                    }
                    return $q.all(catalogs);
                });
        }

        function getIdFromLocation(uri) {
            uri = uri || new URI().toString();
            var parts = uri.split('-');
            return parts[parts.length - 2];
        }

        function resolveUriFromHierarchy(categoryId, locale, channel) {
            return travelUpNavigationHierarchy(categoryId, locale)
                .then(function (catalogs) {
                    return catalogUrlSchema.build(catalogs, channel)
                });
        }

        function getById(categoryId) {
            return menuService.findInNavigation(categoryId)
                .then(function (catalog) {
                    catalog.channel = metaTag.getOcsChannel()
                    return catalog.$get().$promise;
                });
        }

        function travelUpHierarchy(categoryId) {
            return getById(categoryId)
                .then(function (catalog) {
                    var catalogs = [];
                    while (catalog != null) {
                        catalogs.push(catalog);
                        catalog = getById(catalog.parentId, locale);
                    }
                    return $q.all(catalogs);
                });
        }

        function resolveUri(categoryId) {
            return travelUpHierarchy(categoryId)
                .then(catalogUrlSchema.build);
        }

    }

})(angular);
