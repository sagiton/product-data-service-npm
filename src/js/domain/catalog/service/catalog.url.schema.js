(function (angular) {
    angular
        .module('pds.catalog.service')
        .service('catalogUrlSchema', CatalogUrlSchema);

    CatalogUrlSchema.$inject = ['SeoFriendlyUrlBuilder', '_'];

    function CatalogUrlSchema(SeoFriendlyUrlBuilder, _) {
        var productPrefix = 'p';
        var categoryPrefix = 'c';

        var schemas = {
            'comercial-e-industrial': industrialSchema, //TODO Make this locale independent
            'gewerbe-industrie': industrialSchema,
            'commercial-industrial': industrialSchema
        };

        function getSiteChannel() {
            return angular.element('meta[name="channel"]').attr('content') || '';
        }

        this.build = function (catalogs) {
            if (!_.size(catalogs)) {
                return String();
            }
            return getSchema(getSiteChannel())(catalogs);
        };

        function getSchema(name) {
            return schemas[name] || residentialSchema;
        }

        function industrialSchema(catalogs) {
            var builder = new SeoFriendlyUrlBuilder({ocsBasePath: getSiteChannel() + '/ocs'});
            _.forEachRight(catalogs, function (catalog, index) {
                var fragments = [catalog.name];
                if (index === 0) {
                    fragments.push(catalog.id, categoryPrefix);
                }
                builder.addPath(fragments);

                if (catalog.isProductFamily()) {
                    builder.setPath([catalog.name, catalog.id, productPrefix]);
                }
            });
            return builder.build();
        }

        function residentialSchema(catalogs) {
            var builder = new SeoFriendlyUrlBuilder();
            _.forEachRight(catalogs, function (catalog, index) {
                var fragments = [catalog.name];
                if (index === 0) {
                    fragments.push(catalog.id, categoryPrefix);
                }
                if (index === catalogs.length - 1) {
                    fragments.unshift(getSiteChannel());
                }
                builder.addPath(fragments);

                if (catalog.isProductFamily()) {
                    builder.setPath([catalog.name, catalog.id, productPrefix]);
                }
            });
            return builder.build();
        }
    }


})(angular);
