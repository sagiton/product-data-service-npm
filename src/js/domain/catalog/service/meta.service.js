(function (angular) {
    angular
        .module('pds.catalog.service')
        .service('MetaService', MetaService);

    MetaService.$inject = ['$rootScope', '$scope', '$q', '$location', 'CatalogService', 'imageUrlFilter', 'config'];

    var TITLE_DELIMITER = ' | ';
    var LOCALE_DELIMITER = '-';
    var LOCALE_PROPER_DELIMITER = '_';

    function MetaService($rootScope, $scope, $q, $location, CatalogService, imageUrlFilter, config) {
        $scope.$on('pds.catalog.loaded', function (event, params) {
            return updateMetaByCatalog(params.catalog);
        });

        return {
            updateMetaByCatalog: updateMetaByCatalog
        };

        function updateMetaByCatalog(catalog) {
            return CatalogService
                .travelUpNavigationHierarchy(catalog.id.value)
                .then(function (tree) {
                    tree[0] = catalog;
                    return tree;
                })
                .then(function (tree) {
                    var q = $q.defer();
                    var currentNode = tree[0];
                    tree[0].name = tree[0].name.value;
                    var headerTitle = [];

                    for (var i = 0; i < tree.length; i++) {
                        if (tree[i] && tree[i].name) {
                            headerTitle.push(tree[i].name);
                        }
                    }
                    headerTitle.push(config.metaTags.siteName);

                    var image = (currentNode.keyVisual || currentNode.productimage || {}).value;
                    var event = {
                        title: headerTitle.join(TITLE_DELIMITER),
                        description: (currentNode.seoMetaText || currentNode.descriptionLong || currentNode.descriptionShort || {}).value,
                        image: image ? imageUrlFilter(image) : undefined,
                        siteName: config.metaTags.siteName,
                        webTrends: {
                            cg_s: tree[0] ? tree[0].name : null,
                            z_cg3: tree[1] ? tree[1].name : null,
                            z_cg4: tree[2] ? tree[2].name : null
                        },
                        canonicalUrl: $location.absUrl()
                    };

                    //TODO Project specific logic - externalize
                    if (!(currentNode.blockCanonicalTag || {}).value) {
                        var canonicalRef = (currentNode.canonicalRef || {}).value;
                        if(canonicalRef) {
                            CatalogService
                                .resolveUriFromHierarchy(canonicalRef)
                                .then(function (url) {
                                    event.canonicalUrl = url;
                                    q.resolve(event);
                                });
                        } else {
                            q.resolve(event);
                        }
                    }

                    return q.promise;
                })
                .then(function (params) {
                    $rootScope.$broadcast('pds.header.update', params);
                })
                .then(function () {
                    angular
                        .element('link[hreflang]')
                        .each(function (index, link) {
                            var linkObject = angular.element(link);
                            var locale = linkObject.attr('hreflang');
                            locale = locale.replace(LOCALE_DELIMITER, LOCALE_PROPER_DELIMITER);
                            CatalogService
                                .resolveUriFromHierarchy(categoryId, locale)
                                .then(function (url) {
                                    linkObject.attr('href', url);
                                });
                        });
                });
        }
    }
})(angular);
