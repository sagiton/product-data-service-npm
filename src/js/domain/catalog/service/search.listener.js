(function (angular) {
    angular
        .module('pds.catalog.service')
        .service('catalogSearchListener', CatalogSearchListener);

    CatalogSearchListener.$inject = ['$rootScope', '$q', 'config'];

    function CatalogSearchListener($root, $q, config) {
        this.listen = function () {
            var def = $q.defer();
            $root.$on('pds.search.navigate', function (event, params) {
                if (params.target.channelDiscriminator == config.search.pdsChannelDiscriminator || params.target.resourceId) {
                    def.resolve(params);
                }
            });
            return def.promise;
        }
    }
})(angular);
