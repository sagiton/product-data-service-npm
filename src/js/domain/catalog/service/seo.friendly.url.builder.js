(function (angular, URI) {
    angular
        .module('pds.catalog.service')
        .factory('SeoFriendlyUrlBuilder', SeoFriendlyUrlBuilderFactory);

    SeoFriendlyUrlBuilderFactory.$inject = ['$window', '_', 'simplifyCharactersFilter', 'config'];

    var fragmentSeparator = '-';
    var pathSeparator = '/';

    function SeoFriendlyUrlBuilderFactory($window, _, simplifyCharactersFilter, config) {
        function SeoFriendlyUrlBuilder(options) {
            this.path = buildBasePath();
            this.simplifyCharactersFilter = simplifyCharactersFilter;
            this.options = options || {};
        }

        SeoFriendlyUrlBuilder.prototype.addPath = function(fragments) {
            if (!fragments) {
                return this;
            }
            var args = _.compact([].concat(fragments));
            this.path += pathSeparator + this.simplifyCharactersFilter(args.join(fragmentSeparator));
            return this;
        };

        SeoFriendlyUrlBuilder.prototype.setPath = function (fragments) {
            this.path = buildBasePath();
            this.addPath(fragments);
            return this;
        };

        SeoFriendlyUrlBuilder.prototype.build = function () {
            return this.path + (config.urlSchema.trailingSlash ? '/' : '');
        };

        return SeoFriendlyUrlBuilder;

        function buildBasePath() {
            return URI().origin() + $window.getBasePath() + config.pdsPathPrefix;
        }
    }


})(angular, URI);
