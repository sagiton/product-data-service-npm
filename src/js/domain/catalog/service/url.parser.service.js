(function (angular) {

    angular
        .module('pds.catalog.service')
        .service('urlParserService', UrlParser);

    UrlParser.$inject = ['config'];

    var countryMatchIndex = 1;
    var languageMatchIndex = 2;
    var rootSegmentMatchIndex = 3;
    var catalogIdMatchIndex = 4;

                      /**  {1}{2}   {3}                                    {4}    **/
                      /** /ch/de/residential.html/qwe/asd/qwe/asd/poiuuy-134233-c **/
    var pathPattern = /^\/([a-z]{2}|global)\/([a-z]{2})\/(?:ocs\/)?([^\/]*)(?:\.html)?\/(?:.*-([0-9]*)-[pc]\/?$)?/i;
    var catalogPattern = /\/ocs\/(.*)/;

    function UrlParser(config) {
        this.config = config;
    }

    UrlParser.prototype.isOCS = function() {
        return !!this.getCatalogId();
    };

    UrlParser.prototype.getRootSegment = function getRootSegment() {
        return matchForIndex(rootSegmentMatchIndex) || this.config.metaTags.siteName;
    };

    UrlParser.prototype.getCatalogId = function () {
        return matchForIndex(catalogIdMatchIndex);
    };

    UrlParser.prototype.getLanguage = function () {
        return matchForIndex(languageMatchIndex);
    };

    UrlParser.prototype.getCatalogPath = function (url) {
        var uri = new URI();
        if (url) {
            uri = new URI(url);
        }
        var match = uri.path().match(catalogPattern);
        return match && match[1];
    };

    function matchForIndex(index) {
        var match = new URI().path().match(pathPattern);
        return match && match[index];
    }

})(angular);
