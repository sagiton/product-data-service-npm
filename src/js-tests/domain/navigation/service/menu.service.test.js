import * as _ from 'lodash'
import ResourceWrapper from '../../../__mocks__/resource.mock'
import {getMenu} from "../../../__mocks__/api.service.mock";

describe('service: MenuService', () => {
    let MenuService

    beforeEach(angular.mock.module('pds.navigation.service'))
    beforeEach(() => {
        spyOn(angular, 'element').and.returnValue({
            attr: val => 'OCS_CHANNEL',
            data: () => {}
        });

        angular.mock.module($provide => {
            $provide.value('_', _)
            $provide.value('Navigation', new ResourceWrapper(getMenu()))
            $provide.value('locale', jest.fn())
        });
        angular.mock.inject($injector => MenuService = $injector.get('MenuService'))
    });

    it('should return menu', async () => {
        const menu = await MenuService.getMenu();
        expect(menu).toMatchSnapshot()
    })

    it('should find in navigation', async () => {
        const node = await MenuService.findInNavigation('669462');
        const node2 = await MenuService.findInNavigation('669464');
        expect({node, node2}).toMatchSnapshot('nodes')
        expect(MenuService.flatNavigation).toMatchSnapshot('flatMenu')
    })

    it('should find parent in navigation', async () => {
        await MenuService.findInNavigation()
        const node = await MenuService.findParentInNavigation('1106917');
        const node2 = await MenuService.findParentInNavigation('669445');
        expect({node, node2}).toMatchSnapshot()
    })
})
