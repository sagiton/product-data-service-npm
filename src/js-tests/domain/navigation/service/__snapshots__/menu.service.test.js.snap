// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service: MenuService should find parent in navigation 1`] = `
Object {
  "node": undefined,
  "node2": undefined,
}
`;

exports[`service: MenuService should return menu 1`] = `
Object {
  "children": Array [
    Object {
      "children": Array [],
      "id": "669447",
      "name": "Heizkessel",
      "type": "LEAF_CATEGORY",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "id": "1106916",
              "name": "UT-L",
              "type": "PRODUCT",
            },
          ],
          "id": "669463",
          "name": "UNIMAT Heizkessel UT-L",
          "type": "PRODUCT_FAMILY",
        },
        Object {
          "children": Array [
            Object {
              "id": "1106917",
              "name": "UT-M",
              "type": "PRODUCT",
            },
          ],
          "id": "669464",
          "name": "UNIMAT Heißwasserkessel UT-M",
          "type": "PRODUCT_FAMILY",
        },
        Object {
          "children": Array [
            Object {
              "id": "1106918",
              "name": "UT-H",
              "type": "PRODUCT",
            },
          ],
          "id": "669465",
          "name": "UNIMAT Heißwasserkessel UT-H",
          "type": "PRODUCT_FAMILY",
        },
        Object {
          "children": Array [
            Object {
              "id": "1106919",
              "name": "UT-HZ",
              "type": "PRODUCT",
            },
          ],
          "id": "669466",
          "name": "UNIMAT Heißwasserkessel UT-HZ",
          "type": "PRODUCT_FAMILY",
        },
      ],
      "id": "669462",
      "name": "Heißwasserkessel",
      "type": "LEAF_CATEGORY",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "id": "1106758",
              "name": "U-ND",
              "type": "PRODUCT",
            },
            Object {
              "id": "1106759",
              "name": "U-HD",
              "type": "PRODUCT",
            },
          ],
          "id": "669472",
          "name": "UNIVERSAL Dampfkessel U-ND, U-HD",
          "type": "PRODUCT_FAMILY",
        },
        Object {
          "children": Array [
            Object {
              "id": "1106911",
              "name": "U-MB",
              "type": "PRODUCT",
            },
          ],
          "id": "669473",
          "name": "UNIVERSAL Dampfkessel U-MB",
          "type": "PRODUCT_FAMILY",
        },
        Object {
          "children": Array [
            Object {
              "id": "1106913",
              "name": "UL-SX",
              "type": "PRODUCT",
            },
            Object {
              "id": "1106912",
              "name": "UL-S",
              "type": "PRODUCT",
            },
          ],
          "id": "669474",
          "name": "UNIVERSAL Dampfkessel UL-S, UL-SX",
          "type": "PRODUCT_FAMILY",
        },
        Object {
          "children": Array [
            Object {
              "id": "1106914",
              "name": "ZFR",
              "type": "PRODUCT",
            },
            Object {
              "id": "1106915",
              "name": "ZFR-X",
              "type": "PRODUCT",
            },
          ],
          "id": "669475",
          "name": "UNIVERSAL Dampfkessel ZFR, ZFR-X",
          "type": "PRODUCT_FAMILY",
        },
      ],
      "id": "669471",
      "name": "Dampfkessel",
      "type": "LEAF_CATEGORY",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "id": "727826",
              "name": "BHKW-Modul CHP CE 12 NA",
              "type": "PRODUCT",
            },
            Object {
              "id": "762957",
              "name": "BHKW-Modul CHP CE 19-2 NA",
              "type": "PRODUCT",
            },
            Object {
              "id": "1106923",
              "name": "CHP CE 50-400 NA",
              "type": "PRODUCT",
            },
          ],
          "id": "669481",
          "name": "BHKW-Module",
          "type": "PRODUCT_FAMILY",
        },
        Object {
          "children": Array [
            Object {
              "id": "1107093",
              "name": "BHKW-Systeme",
              "type": "PRODUCT",
            },
          ],
          "id": "669482",
          "name": "BHKW-Systeme",
          "type": "PRODUCT_FAMILY",
        },
        Object {
          "children": Array [
            Object {
              "id": "1107094",
              "name": "Druckluft-Wärme-Kraftwerk",
              "type": "PRODUCT",
            },
          ],
          "id": "669483",
          "name": "Druckluft-Wärme-Kraftwerk",
          "type": "PRODUCT_FAMILY",
        },
      ],
      "id": "669480",
      "name": "Blockheizkraftwerke",
      "type": "LEAF_CATEGORY",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "children": Array [
                Object {
                  "id": "1107117",
                  "name": "HHR // Hochtemperaturwärmepumpen und Kältemaschinen HHR",
                  "type": "PRODUCT",
                },
              ],
              "id": "669518",
              "name": "Hochtemperatur-Wärmepumpen und Kältemaschinen HHR",
              "type": "PRODUCT_FAMILY",
            },
          ],
          "id": "669517",
          "name": "Großwärmepumpen",
          "type": "LEAF_CATEGORY",
        },
      ],
      "id": "669484",
      "name": "Kühlung & Lüftung",
      "type": "CATEGORY",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "id": "1107125",
              "name": "Eigenbefeuerter Abhitzekessel",
              "type": "PRODUCT",
            },
          ],
          "id": "669520",
          "name": "4-Zug Abhitzekessel",
          "type": "PRODUCT_FAMILY",
        },
        Object {
          "children": Array [
            Object {
              "id": "1106924",
              "name": "UNIVERSAL Abhitzedampfkessel HRSB",
              "type": "PRODUCT",
            },
          ],
          "id": "669521",
          "name": "UNIVERSAL Abhitzedampfkessel",
          "type": "PRODUCT_FAMILY",
        },
        Object {
          "children": Array [
            Object {
              "id": "1107126",
              "name": "Abhitzekessel",
              "type": "PRODUCT",
            },
          ],
          "id": "669522",
          "name": "Abhitzekessel",
          "type": "PRODUCT_FAMILY",
        },
        Object {
          "children": Array [
            Object {
              "id": "1107127",
              "name": "ORC-Systeme",
              "type": "PRODUCT",
            },
          ],
          "id": "669523",
          "name": "ORC-Systeme",
          "type": "PRODUCT_FAMILY",
        },
      ],
      "id": "669519",
      "name": "Abhitzesysteme",
      "type": "LEAF_CATEGORY",
    },
    Object {
      "children": Array [
        Object {
          "children": Array [
            Object {
              "id": "1107087",
              "name": "Solarthermische Großanlagen",
              "type": "PRODUCT",
            },
          ],
          "id": "669460",
          "name": "Solarthermische Großanlagen",
          "type": "PRODUCT_FAMILY",
        },
      ],
      "id": "732005",
      "name": "Trinkwassererwärmung",
      "type": "LEAF_CATEGORY",
    },
    Object {
      "children": Array [],
      "id": "669524",
      "name": "Komponenten",
      "type": "CATEGORY",
    },
  ],
  "id": "669446",
  "maxNavigationItems": 6,
  "name": "Produkte",
  "type": "CATEGORY",
}
`;
